File: language-support/js/daml-ledger/src/validation/get_transaction_by_event_id_request.ts
// Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

import * as ledger from "..";
import { array, native, object, Validation } from "./base";
import { RequiredValidation } from "./base/typelevel";

function required(): RequiredValidation<ledger.GetTransactionByEventIdRequest> {
    return {
        eventId: native('string'),
        requestingParties: array(native('string')),
    };
}

export const GetTransactionByEventIdRequest: Validation = object<ledger.GetTransactionByEventIdRequest>('GetTransactionByEventIdRequest', required, () => ({}));

File: ledger/ledger-api-domain/src/main/scala/com/digitalasset/ledger/api/messages/transaction/GetTransactionByEventIdRequest.scala
// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
// SPDX-License-Identifier: Apache-2.0

package com.daml.ledger.api.messages.transaction

import com.daml.lf.data.Ref.Party
import com.daml.ledger.api.domain.{EventId, LedgerId}

final case class GetTransactionByEventIdRequest(
    ledgerId: Option[LedgerId],
    eventId: EventId,
    requestingParties: Set[Party],
)
